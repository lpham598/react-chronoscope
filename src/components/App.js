"use strict";
exports.__esModule = true;
var React = require("react");
var TreeGraph_1 = require("./TreeGraph");
var LineGraph_1 = require("./LineGraph");
// export interface HelloProps { compiler: string; framework: string; };
exports.App = function ( /*props: HelloProps*/) {
    return (
    // <div>
    //   {/* <Hello /> */}
    //   <h1>Hello from {props.compiler} and {props.framework}!</h1>
    //   <div id='treeGraphDiv'>
    //     <TreeGraph/>
    //   </div>
    //   <div id='lineGraphDiv'>
    //     <LineGraph/>
    //   </div>
    //   <h2 style={{width: '10px'}}>Test Test</h2>
    // </div>
    <div>
    
      <h1 style={{ textAlign: "center", fontSize: "28px" }}>React ChronoScope</h1>
      <hr style={{ border: "2px solid black" }}></hr>
      <h2>Tree Diagram</h2>
      <div id='treeGraphDiv' style={{ backgroundColor: "whitesmoke" }}>
        <TreeGraph_1["default"] />
      </div>
      <hr style={{ border: "2px solid black" }}></hr>
      <div id='lineGraphDiv'>
        <h2>TimeLine</h2>
        <LineGraph_1["default"] />
      </div>
    </div>);
};
