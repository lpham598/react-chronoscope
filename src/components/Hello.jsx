import React, { useState, useEffect } from 'react';

export const Hello = (props) => {
    const [greeting, setGreeting] = useState('Hello World');

    useEffect(() => {
        console.log('UseEffect triggered');
    });

    return (
        <div>
            <h1>{greeting}</h1>,
            <button onClick={() => setGreeting(greeting + 'Hahahaha')}>Change Greeting</button>
        </div>
    )
};