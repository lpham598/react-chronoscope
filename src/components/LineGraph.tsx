import * as React from 'react';
import Timeline from 'react-visjs-timeline';

const options = {
  width: '100%',
  height: '250px',
  stack: true,
  showCurrentTime: false,
  showMajorLabels: false,
  zoomable: true,
  start: new Number(305),
  end: new Number(320),
  min: new Number(0),
  max: new Number(1000),
  type: 'range',
  selectable: true,
  // horizontalScroll : true,
  verticalScroll: true,
  // editable: true,
};

interface event {
  title: string,
  content: string,
  start: any,
  end: any,
}

const items = [];
let event;
let tempData;

function getData(Node) {
  event = {};
  event.start = new Number(Math.floor(Node.attributes.renderStart));
  event.end = new Number(Math.floor(Node.attributes.renderStart + Node.attributes.renderTotal));
  event.content = Node.name;
  event.title = Node.name;
  items.push(event);
  if (Node.children.length !== 0) {
    Node.children.forEach((child) => {
      getData(child);
    });
  }
}
interface State {
  data: object[],
  render: boolean
}

class LineGraph extends React.Component <{}, State> {
  state: State = {
    data: items,
    render: false,
  };

  componentDidMount() {
    // open connection with background script
    const port = chrome.runtime.connect();
    // listen for a message from the background script
    port.onMessage.addListener((message) => {
      setTimeout(() => {
        getData(message.payload.payload.children[0]);
        this.setState({
          data: items,
          render: true,
        });
      }, 200);
      // abort connection
      port.disconnect();
    });
  }

  // componentDidUpdate() {
  //   setTimeout(() => {
  //     // open connection with background script
  //     // const port = chrome.runtime.connect();
  //     // port.onMessage.addListener(message => {
  //     //   items = [];
  //     //   tempData = getData(message.payload.payload.children[0]);
  //     //   if (JSON.stringify(tempData) !== JSON.stringify(items)) {
  //     //     items = tempData;
  //     //     console.log('Timeline componentDidUpdate: ', items);
  //         console.log('Updated state to: ', this.state.data);
  //         this.setState({
  //           render: false
  //         });
  //         this.setState({
  //           data: items,
  //           render: true
  //         });
  //   //     }
  //   //   })
  //   }, 20000);
  // }

  render() {
    return (
      <div id="lineGraph" style={{}}>
        {this.state.render
            && (
            <Timeline
              options={options}
              items={this.state.data}
            />
            )}
      </div>
    );
  }
}

export default LineGraph;
