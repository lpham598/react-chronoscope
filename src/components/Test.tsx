import * as React from "react";

  
  class Test extends React.Component {
    render() {
      return (
        // <div id="treeWrapper" style={{width: '50em', height: '20em'}}>
        <div>

        <h1> test test </h1>
          
  
        </div>
      );
    }
  }

export default Test;