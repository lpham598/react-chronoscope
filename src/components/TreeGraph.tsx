import * as React from "react";
import Tree from 'react-d3-tree';

  
const myTreeData = [
  {
    name: 'App',
    attributes: {
      keyApp: 'val A',
    },
    children: [
      {
        name: 'A1',
        attributes: {
          keyA: 'val A',
        },
      },
      {
        name: 'A2',
      },
    ],
  },
];

  class TreeGraph extends React.Component {
    render() {
      return (
        // <div id="treeWrapper" style={{width: '50em', height: '20em'}}>
        <div id='treeGraph' style={{}}>

        <Tree data={myTreeData} orientation="vertical" translate={{ x: 100, y: 20}}/>
  
        </div>
      );
    }
  }

export default TreeGraph;