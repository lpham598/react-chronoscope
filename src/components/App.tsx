import * as React from "react";
import TreeGraph from "./TreeGraph";
import LineGraph from "./LineGraph"

export interface HelloProps { compiler: string; framework: string; };

export const App = (props: HelloProps) => {
  return (
    // <div>
    //   {/* <Hello /> */}
    //   <h1>Hello from {props.compiler} and {props.framework}!</h1>
    //   <div id='treeGraphDiv'>
    //     <TreeGraph/>
    //   </div>
    //   <div id='lineGraphDiv'>
    //     <LineGraph/>
    //   </div>
    //   <h2 style={{width: '10px'}}>Test Test</h2>
    // </div>
    <div>
    {/* <img src="ChronoScope.png"></img> */}
      <h1 style={{textAlign : "center", fontSize: "28px"}}>React ChronoScope</h1>
      <hr style={{border: "2px solid black"}}></hr>
      <h2>Tree Diagram</h2>
      <div id='treeGraphDiv' style={{backgroundColor:"whitesmoke"}}>
        <TreeGraph/>
      </div>
      <hr style={{border: "2px solid black"}}></hr>
      <div id='lineGraphDiv'>
        <h2>TimeLine</h2>
        <LineGraph/>
      </div>
    </div>
    )
}

