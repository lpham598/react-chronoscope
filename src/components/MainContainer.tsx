import * as React from 'react';
import { useState, useEffect } from 'react';
import TreeGraph from './TreeGraph';
import LineGraph from './LineGraph';
import { ITree } from '../interfaces';

let treeGraphData: ITree[] = [{
  name: '',
  children: [],
}];

// initialize port that will be upon when component Mounts
let port;

const timeLineArray = [];

let items = [];

const options = {
  width: '100%',
  height: '500px',
  stack: true,
  showCurrentTime: false,
  showMajorLabels: false,
  zoomable: false,
  start: new Number(0),
  end: new Number(10),
  min: new Number(0),
  max: new Number(40),
  type: 'range',
  selectable: true,
  horizontalScroll: false,
  verticalScroll: true,
  // editable: true,
};

let event;

interface Event {
    title: string,
    content: string,
    start: any,
    end: any,
}

function getData(Node, baseTime) {
  event = {};
  event.start = new Number((Number(Node.attributes.renderStart) - Number(baseTime)).toFixed(2));
  event.end = new Number((Number(Node.attributes.renderStart) + Number(Node.attributes.renderTotal) - Number(baseTime)).toFixed(2));
  console.log('render start: ', Number(Node.attributes.renderStart));
  console.log('rendered total: ', Number(Node.attributes.renderTotal));
  console.log('base time: ', Number(baseTime));
  console.log('final Number Object', new Number(Number(Node.attributes.renderStart) + Number(Node.attributes.renderTotal) - Number(baseTime)).toFixed(2));
  event.content = Node.name;
  event.title = Node.name;
  items.push(event);
  if (Node.children.length !== 0) {
    Node.children.forEach((child) => {
      getData(child, baseTime);
    });
  }
}

export const MainContainer: React.FC = () => {
  const [tree, setTree] = useState<ITree[]>(treeGraphData);

  useEffect(() => {
    // open connection with background script
    // make sure to open only one port
    if (!port) port = chrome.runtime.connect();
    // listen for a message from the background script
    port.onMessage.addListener((message) => {
      if (JSON.stringify([message.payload.payload]) !== JSON.stringify(treeGraphData)) {
        // save new tree
        treeGraphData = [message.payload.payload];
        setTree(treeGraphData);
        getData(treeGraphData[0], treeGraphData[0].attributes.renderStart);
        timeLineArray.shift();
        timeLineArray.push(items);
        items = [];
      }
    });
  });

  return (
    <>
      <h1 style={{ textAlign: 'center', fontSize: '28px' }}>React ChronoScope</h1>
      <hr style={{ border: '2px solid black' }} />
      <h2>Tree Diagram</h2>
      <div id="treeGraphDiv" style={{ backgroundColor: 'whitesmoke' }}>
        <TreeGraph data={tree} />
      </div>
      <hr style={{ border: '2px solid black' }} />
      <div id="lineGraphDiv">
        <h2>TimeLine</h2>
        {
            timeLineArray.map((el, i, arr) => {
              return <LineGraph data={el} options={options} />;
            })
        }
      </div>
    </>
  );
};
