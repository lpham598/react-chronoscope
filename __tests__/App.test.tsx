import React from 'react';
import { shallow } from 'enzyme';

import { App } from '../src/components/App';

describe('React-ChronoScope Component Tests', () => {
  describe('Component: App', () => {
    let wrapper;

    beforeAll(() => {
      wrapper = shallow(<App/>);
    });

    it('should render correctly with no props', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('Renders a <div> tag', () => {
      expect(wrapper.type()).toEqual('div');
    });
  });
});