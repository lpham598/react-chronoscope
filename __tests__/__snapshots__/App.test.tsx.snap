// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React-ChronoScope Component Tests Component: App should render correctly with no props 1`] = `
<div>
  <h1
    style={
      Object {
        "fontSize": "28px",
        "textAlign": "center",
      }
    }
  >
    React ChronoScope
  </h1>
  <hr
    style={
      Object {
        "border": "2px solid black",
      }
    }
  />
  <h2>
    Tree Diagram
  </h2>
  <div
    id="treeGraphDiv"
    style={
      Object {
        "backgroundColor": "whitesmoke",
      }
    }
  >
    <TreeGraph />
  </div>
  <hr
    style={
      Object {
        "border": "2px solid black",
      }
    }
  />
  <div
    id="lineGraphDiv"
  >
    <h2>
      TimeLine
    </h2>
    <LineGraph />
  </div>
</div>
`;
